parser/parser.go

```diff
@@ -245,6 +250,64 @@ func parseUnary(p *Parser) (ast.Node, error) {
 	return ast.NewUnaryExpr(tok.Type, expr), nil
 }
 
+func parseVarDecls(p *Parser) (ast.Node, error) {
+	p.forget(1)
+	return _parseVarDecls(p)
+}
+
+func _parseVarDecls(p *Parser) (ast.VarDecls, error) {
```

Comment:
o.o

Reply:
This is because I wanted to recursively build the vardecls from other vardecls but to use the function on the parserfn maps it needs to return an ast.Node. If I allow VarDecls to be built from ast.Node I would allow all sort of odd stuff to happen like var decls with other things inside (function declarations for example).

Reply:
I'm complaining mostly about the name hehe